File: java/Majority_element.java
public class Majority_element {
    public static void main(String[] args) {
        int[] nums = {2, 7, 11, 15};
        int target = 9;
        Solution sol = new Solution();
        sol.twoSum(nums, target);
    }
}

class Solution {
    public int[] twoSum(int[] nums, int target) {

        int add = 0, n = nums.length;
        int i = 0;

        while (i < n - 1) { // Prevent out-of-bounds
            add = nums[i] + nums[i + 1];
            if (add == target) {
                System.out.print(nums[i]);
                System.out.println(nums[i + 1]);
            }
            i++;
        }
        return null; 
    }
}

File: c++/Array/majorityelement.cpp
#include <iostream>
#include <vector>
using namespace std;


class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0;
        int candidate = 0;

        //finding the potential candidate
        for (int num : nums) {
            if (count == 0)
                candidate = num;
            if (num == candidate)
                count++;
            else
                count--;
        }

       //Verifying the candidate
        int freq = 0;
        for (int num : nums) {
            if (num == candidate)
                freq++;
        }

        // If needed, verify it's truly majority
        if (freq > nums.size() / 2)
            return candidate;
        else
            return -1; // no majority element found
    }
};

int main() {
    Solution sol;

    // example 
    vector<int> nums = {2, 2, 1, 1, 1, 2, 2};

    int result = sol.majorityElement(nums);

    if (result != -1)
        cout << "Majority Element: " << result << endl;
    else
        cout << "No Majority Element found." << endl;

    return 0;
}
